src/status_im/contexts/wallet/create_account/view.cljs

```diff
@@ -50,23 +51,44 @@
 
 (defn- f-view
   []
-  (let [top                          (safe-area/get-top)
-        bottom                       (safe-area/get-bottom)
-        account-color                (reagent/atom (rand-nth colors/account-colors))
-        emoji                        (reagent/atom (emoji-picker.utils/random-emoji))
-        number-of-accounts           (count (rf/sub [:wallet/accounts-without-watched-accounts]))
-        account-name                 (reagent/atom "")
-        placeholder                  (i18n/label :t/default-account-placeholder
-                                                 {:number (inc number-of-accounts)})
-        derivation-path              (reagent/atom (utils/get-derivation-path number-of-accounts))
-        {:keys [public-key address]} (rf/sub [:profile/profile])
-        on-change-text               #(reset! account-name %)
-        primary-name                 (first (rf/sub [:contacts/contact-two-names-by-identity
-                                                     public-key]))
-        {window-width :width}        (rn/get-window)]
+  (let [top                                   (safe-area/get-top)
+        bottom                                (safe-area/get-bottom)
+        account-color                         (reagent/atom (rand-nth colors/account-colors))
+        emoji                                 (reagent/atom (emoji-picker.utils/random-emoji))
+        number-of-accounts                    (count (rf/sub
+                                                      [:wallet/accounts-without-watched-accounts]))
+        account-name                          (reagent/atom "")
+        placeholder                           (i18n/label :t/default-account-placeholder
+                                                          {:number (inc number-of-accounts)})
+        derivation-path                       (reagent/atom (utils/get-derivation-path
+                                                             number-of-accounts))
+        {:keys [address customization-color]} (rf/sub [:profile/profile])
+        on-change-text                        #(reset! account-name %)
+        {window-width :width}                 (rn/get-window)]
     (fn [{:keys [theme]}]
-      (let [{:keys [new-keypair]} (rf/sub [:wallet/create-account])]
-        (rn/use-unmount #(rf/dispatch [:wallet/clear-new-keypair]))
+      (let [{:keys [new-keypair]}           (rf/sub [:wallet/create-account])
+            keypairs                        (rf/sub [:wallet/keypairs])
+            selected-keypair-uid            (rf/sub [:wallet/selected-keypair-uid])
+            keypair                         (first (filter #(= (:key-uid %) selected-keypair-uid)
+                                                           keypairs))
```

Comment:
Consider using `some` instead of `first` + `filter`

Reply:
@ulisesmac Some doesn't seem to work here. It returns `true` instead of the keypair value

Reply:
@OmarBasem yeah, you need to return the value when the test is true.

```clojure
(some #(when (= (:key-uid %) selected-keypair-uid)
         %)
  keypairs)
```